_posts/2019-12-08-what-is-a-shell.md

```diff
@@ -0,0 +1,166 @@
+Aside from the cross-platform aspect of Nushell,
+one of the more appealing features is how Nushell treats data.
+Instead of treating all data as raw text,
+Nushell takes some inspiration from Powershell and treats data as structured tables or objects.
+This enables Nushell to support a rich command and plugin system to manipulate data in very useful and declarative ways.
+
+### Nushell Examples
+I won't go over the details for how to get started, 
+because [Nushell already offers](https://book.nushell.sh/) a helpful book.
+
+We can start off with the typical `ls` example.
+Nushell offers its own version of `ls` which you can learn about by using the help command `help ls`.
+Using only `ls` you'll see a table returned to you with some nice default information.
+Do you want to find all the symlinks in your `/usr/bin` directory?
+`ls /usr/bin | where type == Symlink`.
+Okay, now what about all the symlinks in your `C:\Users` directory on Windows?
+`ls C:\Users | where type == Symlink`.
+The same thing? Well, the same command aside from the directory... This is Great!
+
+Once you invest the time into learning Nushell,
+you'll be able to use the same commands it on any system that is running it.
+Our example above using `ls` is actually using the `ls` command that is inside of Nushell and not the "native" `ls` on your system.
+If you prefer the version your OS has, you can always escape the command by prefixing it with `^`, as in `^ls`.
+From there Nushell can still take advantage of the output by piping the output of the native command into other Nushell commands.
+
+As mentioned earlier, Nushell is still quite early in its development.
+While it has [many useful features](https://www.nushell.sh/documentation.html), it does not yet support scripting or aliasing of commands.
+As Nushell matures, many of the standard behaviors you might expect from a shell will be included ... and maybe more!?
+
+
+Nushell offers a plugin system in addition to its commands.
+A Nushell plugin is meant to be optional and can be installed in addition to the "core" of Nushell.
+Plugins can be written in many languages as long as they adhere to the plugin interface.
+There already exist plugins written in Rust, Ruby, Python, and more!
+Plugins can be offered as operating system specific programs like something that interacts with an OS specific API or even augment existing programs like SQLite, which is already supported!
+
+My daily workflow involves testing REST APIs.
+I am able to use the `fetch` command offered by nushell to download data and turn it into any format I want.
+In addition to this, I can take files in one format, convert to another and then `post` to a url.
+I frequently have to take CSV files and send them to an endpoint.
+With Nushell I can open a CSV file,
+turn it into a JSON payload,
+and send it all in one line.
+There are examples of this in the [Nushell Cookbook](https://github.com/nushell/cookbook#http)
```

Comment:
You may want to move the cookbook after your example, and tee up the example with something like "Let's see how this works in practice"